### PR TITLE
build: Rename `clib` binary name

### LINKED ISSUE #3800: Build output filename collision
### What's up?

I coincidentally saw this warning, which only seems to happen on windows (??) at https://github.com/PRQL/prql/actions/runs/6855982279/job/18642276364

```
warning: output filename collision.
The lib target `prqlc` in package `prqlc-clib v0.10.1 (D:\a\prql\prql\prqlc\bindings\clib)` has the same output filename as the bin target `prqlc` in package `prqlc v0.10.1 (D:\a\prql\prql\prqlc\prqlc)`.
Colliding filename is: D:\a\prql\prql\target\x86_64-pc-windows-msvc\debug\prqlc.pdb
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
```

Should we name the `clib` output to `prqlc-lib`?

### PR DESCRIPTION
**Pull request by max-sixty**
Closes #3800

### CODE REVIEW

File: prqlc/bindings/clib/Cargo.toml [hunk 1]
```diff
@@ -9,9 +9,9 @@ rust-version.workspace = true
 version.workspace = true
 
 [lib]
-name = "prqlc"
 crate_type = ["staticlib", "cdylib"]
 doctest = false
+name = "prqlc_lib"
```
Comment (eitsupi) [inline prqlc/bindings/clib/Cargo.toml:14]: Just a question: why `_lib`, not `_clib`?

Comment (max-sixty) [reply prqlc/bindings/clib/Cargo.toml:14]: No view! 

@aljazerzen what would you prefer?

Comment (aljazerzen) [reply prqlc/bindings/clib/Cargo.toml:14]: I guess `_lib`? I don't know...

**Comment by max-sixty** [comment]
It'll auto-merge, but feel free to pause if we want to decide on the name. It's also very easy to change now we know where all the locations are!

**Pull request merged**